solr/core/src/java/org/apache/solr/util/numeric/DynamicMap.java

```diff
@@ -17,13 +17,19 @@
 
 package org.apache.solr.util.numeric;
 
-import java.util.function.Consumer;
-import java.util.function.IntConsumer;
+import com.carrotsearch.hppc.HashContainers;
 
-public interface IntIntMap {
-  void set(int key, int value);
-  int get(int key);
-  void forEachValue(IntConsumer consumer);
-  void remove(int key);
-  int size();
+public interface DynamicMap {
+  static boolean useArrayBased(int expectedMaxSize) {
+    // for small data set, prefer using array based
+    return expectedMaxSize < (1 << 12);
```

Comment:
if assertions are enabled, then perhaps return false so as to have tests test the DynamicMap.  I've seen @yonik use this technique in data structure sizing and thought it was clever.

Reply:
+1